agent/src/main/java/com/code_intelligence/jazzer/runtime/UnsafeProvider.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.code_intelligence.jazzer.runtime;
+
+import java.lang.reflect.Field;
+import sun.misc.Unsafe;
+
+public final class UnsafeProvider {
+  private static Unsafe unsafe = null;
+
+  public static Unsafe getUnsafe() {
+    if (unsafe == null) {
```

Comment:
Why not skip the "double checked locking" pattern and directly use `getUnsafeInternal` to set the value of `unsafe` once during startup? Is there a special reason to initialize it lazily.

Reply:
I initially included this as I wanted control over when exceptions would be thrown, but since we are now catching `Throwable`, that serves no purpose anymore. Thanks!